Automattic/wp-calypso
# Manage Plugins popup: Avoid doing one request per site

Fixes https://github.com/Automattic/dotcom-forge/issues/8082

## Proposed Changes

* Avoid checking the connection status for every site in the list 
* Filter out the list of sites according to some additional criteria:
  * Sites that return the `feature` to install plugins
  * Sites that return false to `isSiteWPForTeams` and `isSiteP2Hub`

In the following video, we can see a comparison with production. Production on the left, these changes on the right:

https://github.com/Automattic/wp-calypso/assets/3519124/ae33a6ab-6629-498c-845c-5316b653d9af




## Why are these changes being made?


* Doing so many requests is very slow and very resource-intensive because it renders the whole dialog for every request and response. This also gets worse the more sites you have.

## Testing Instructions



* Using a user with multiple sites, apply this patch and navigate to a plugin, e.g. `/plugins/booking`
* On the right hand side, click on the `Manage plugins` CTA and check the time it takes to open the Dialog

## Pre-merge Checklist
